packages/react-magma-dom/src/components/CharacterCounter/CharacterCounter.tsx

```diff
@@ -0,0 +1,107 @@
+import * as React from 'react';
+import styled from '../../theme/styled';
+import { FormFieldContainerProps } from '../FormFieldContainer';
+import { ThemeContext } from '../../theme/ThemeContext';
+import { ThemeInterface } from '../../theme/magma';
+import { I18nContext } from '../../i18n';
+import { InputMessage } from '../Input/InputMessage';
+
+export interface CharacterCounterProps
+  extends Omit<FormFieldContainerProps, 'fieldId'>,
+    React.HTMLAttributes<HTMLDivElement> {
+  /**
+   * Total number of characters in an input.
+   */
+  inputLength?: number;
+  /**
+   * Sets the maximum amount of characters allowed.
+   */
+  maxLength?: number;
+  testId?: string;
+  /**
+   * @internal
+   */
+  theme?: ThemeInterface;
+}
+
+function buildFontWeight(props: CharacterCounterProps) {
+  if (
+    (props.inputLength < props.maxLength && props.inputLength >= 1) ||
+    props.inputLength === props.maxLength
+  ) {
+    return props.theme.typographyVisualStyles.headingXSmall.fontWeight;
+  }
+}
+const StyledWrapper = styled.div``;
```

Comment:
This doesn't seem to be used. Can it be removed?

Reply:
I'm using this for the JSX `<StyledWrapper data-testid={testId} aria-live="polite">`, should it be done another way?

Reply:
It's not actually styled though, so why not just use `<div data-testid={testId} aria-live="polite">`?

Reply:
On it!